src/core/index.ts

```diff
@@ -308,6 +328,25 @@ export class DevAssist {
 				};
 				switch (block.name) {
 					case "read_file": {
+						// Add the permission logic
+						this.providerRef.deref()?.postMessageToWebview({
+							type: "requestPermission",
+							message: "DevAssist needs permission to read the file. Do you want to proceed?",
+							permissionType: "read_file",
+							params: block.params,
+						});
+
+						await pWaitFor(() => this.isToolUsePermissionRecieved, { interval: 100 });
+						if (this.checkIfToolUsePermissionDenied()) {
+							this.userMessageContent.push({
+								type: "text",
+								text: `Permission denied for tool use: ${block.name}`,
+							});
+							this.resetToolUsePermission();
+							break;
+						}
+						this.resetToolUsePermission();
+
```

Comment:
This addition of ask permission is repeated in both the places. It would be better if we place it in a function with appropriate parameters. 

While deploying also we need to ask for permission. Hence i will convert this as a function in my PR.